src/components/banner-component.jsx

```diff
@@ -11,10 +20,21 @@ const OverlayWrapper = ({ condition, wrapper, children }) =>
 
 const BannerComponent = (props) => {
   const nav_height = document.getElementById("app_navbar")?.offsetHeight;
+  const navigate = useNavigate();
 
   onCleanup(() => {
     setBannerMessage(null);
+    setActionUrl("");
+    setAction("");
+    setButtonText("Ok");
   });
+
+  const onClickConfirm = (action) => {
+    if (action === "redirect") {
```

Comment:
May be the function be executed on button click itself can be passed as props? Then u need not worry about the type of click action

Reply:
updated